corehq/apps/app_manager/static/app_manager/js/forms/advanced/case_config_ui.js

```diff
@@ -293,13 +293,6 @@ hqDefine('app_manager/js/forms/advanced/case_config_ui', function () {
                     action.case_properties.push(caseProperty.wrap(p, action));
                 });
 
-                // needed for compatibility with shared templates
-                action.searchAndFilter = false;
-                action.visible_case_properties = ko.computed(function () {
-                    return action.case_properties();
-                });
-                action.saveOnlyEditedFormFieldsEnabled = toggles.toggleEnabled("SAVE_ONLY_EDITED_FORM_FIELDS");
```

Comment:
I can admit I wasn't even aware of the importance of the `actions.js` file. But having these properties here instead of the actions file is what worked for my local testing and for QA testing on staging. Do we know how the `actions.js` file is different from the `case_config_ui.js` file?

Reply:
I think `actions.js` is for "opening a case" and `case_config_ui.js` is for "loading a case". So we might want to keep this code instead of deleting. I will do more testing to make sure.

Reply:
Ok, got it, and having these properties in only the  `case_config_ui.js` file allowed loading a case to work but not "opening a case".

Reply:
> Ok, got it, and having these properties in only the case_config_ui.js file allowed loading a case to work but not "opening a case".

No, this isn't right. Having these properties only in `case_config_ui.js` meant they were available for existing actions (those already present in the app before loading the page) but not for new actions. The `caseConfig` model, the view that encompasses the whole case management area, has a `load_update_cases` attribute and an `open_cases` attribute, each of which is an observable array, that hold the form's update actions and open actions, respectively.

The old code happened was part of the initialization logic for `load_update_cases` and `open_cases`: the pre-existing actions would get wrapped and would have searchAndFilter, visible_case_properties, and saveOnlyEditedFormFieldsEnabled added to them. But **new** actions wouldn't get these properties added to them, which caused js errors because the HTML templates reference these attributes. The logic really belongs in the action definitions themselves, which is where Simon moved it.

This is basically a pre-existing bug with searchAndFilter and visible_case_properties, and you just added to it by following the existing code pattern (which was the reasonable thing to do). Advanced modules aren't used by all that many projects aren't used by a ton of projects, so bugs can go undetected for a while.

Reply:
Great, thank you for testing it out. Sounds like this PR is good to go.